src/clr-angular/utils/focus-trap/focus-trap.directive.ts

```diff
@@ -13,50 +13,99 @@ import {
   Injector,
   OnDestroy,
   PLATFORM_ID,
+  Renderer2,
 } from '@angular/core';
 
 import { FocusTrapTracker } from './focus-trap-tracker.service';
 
 @Directive({ selector: '[clrFocusTrap]' })
 export class FocusTrapDirective implements AfterViewInit, OnDestroy {
-  private _previousActiveElement: HTMLElement;
-  /* tslint:disable-next-line:no-unused-variable */
+  private previousActiveElement: any;
   private document: Document;
 
+  private topReboundEl: any;
+  private bottomReboundEl: any;
+
   constructor(
-    public elementRef: ElementRef,
-    injector: Injector,
+    private el: ElementRef,
+    private injector: Injector,
     private focusTrapsTracker: FocusTrapTracker,
+    private renderer: Renderer2,
     @Inject(PLATFORM_ID) private platformId: Object
   ) {
-    this.document = injector.get(DOCUMENT);
+    this.document = this.injector.get(DOCUMENT);
     this.focusTrapsTracker.current = this;
   }
 
   @HostListener('document:focusin', ['$event'])
   onFocusIn(event: any) {
-    const nativeElement: HTMLElement = this.elementRef.nativeElement;
+    const nativeElement: HTMLElement = this.el.nativeElement;
 
-    if (this.focusTrapsTracker.current === this && !nativeElement.contains(event.target)) {
+    if (this.focusTrapsTracker.current === this && event.target && !nativeElement.contains(event.target)) {
```

Comment:
In what case is `event.target` not set?

Reply:
It's mostly for testing cases. We had a false positive test where we created a custom event, but we didn't specify `its event.target`. 
Once, that custom event is passed into `onFocusIn` method, `nativeElement.contains(even.target)` was returning `false` as of course event.target is `undefined` there. As such it satisfies this conditional check:

```
if (this.focusTrapsTracker.current === this && event.target && !nativeElement.contains(event.target))
```